File: python2.7/handler.py
import json
import logging
import time

logger = logging.getLogger()
logger.setLevel(logging.INFO)

def build_response_body(event, context):
    if event['action'] == 'warmup':
        minimum_duration_milliseconds = event['minimumDurationMilliseconds']
        logger.info('sleeping for {}ms'.format(minimum_duration_milliseconds))
        time.sleep(minimum_duration_milliseconds / 1000.0)
        logger.info('sleeping done')
        return {
            "uptime": open('/proc/uptime', 'r').read(),
            "logStreamName": context.log_stream_name
        }
    return {
        "message": "Go Serverless v1.0! Your function executed successfully!",
        "input": event
    }

def handle_request(event, context):
    logger.info('received {}'.format(event))
    return build_response_body(event, context)
    # Or as lambda-proxy response for API Gateway:
    # return {
    #     "statusCode": 200,
    #     "body": json.dumps(build_response_body(event, context))
    # }

File: nodejs4.3/handler.js
'use strict';

const fs = require('fs');

const buildResponseBody = function(event, context, responseCallback) {
  if (event.action === 'warmup') {
    console.log('sleeping for ' + event.minimumDurationMilliseconds + 'ms');
    setTimeout(function() {
      console.log('sleeping done');
      responseCallback({
        uptime: fs.readFileSync('/proc/uptime', { encoding: 'utf-8' }),
        logStreamName: context.logStreamName
      });
    }, event.minimumDurationMilliseconds);
    return;
  }
  responseCallback({
    message: 'Go Serverless v1.0! Your function executed successfully!',
    input: event,
  });
};

module.exports.handleRequest = (event, context, callback) => {
  console.log('received', event);
  buildResponseBody(event, context, function(responseBody) {
    callback(null, responseBody);
    // Or as lambda-proxy response for API Gateway:
    // callback(null, {
    //   statusCode: 200,
    //   body: JSON.stringify(responseBody),
    // });
  });
};
